explorer/common/trace_stream.h

```diff
@@ -23,38 +43,68 @@ namespace Carbon {
 // is fully treated as a separate file, we should be able to take a different
 // approach where the caller explicitly toggles tracing when switching file
 // contexts.
+
 class TraceStream {
  public:
   // Returns true if tracing is currently enabled.
   auto is_enabled() const -> bool {
-    return stream_.has_value() && !in_prelude_;
+    return stream_.has_value() &&
+           current_code_context_ != CodeContext::Prelude &&
+           (allowed_phases_.count(current_phase_) > 0 ||
+            allowed_phases_.count(ProgramPhase::All) > 0 ||
+            current_phase_ == ProgramPhase::Other);
```

Comment:
Oh yes I we can remove it. I'm sorry I didn't notice it before.
Also I'm thinking of renaming it to `Unknown`, it will feel more intuitive as it represents the unknown phase.